src/traffic_cop/traffic_cop.cpp

```diff
@@ -281,15 +275,23 @@ void TrafficCop::ExecuteStatementPlanGetResult() {
   }
 }
 
+/*
+ * Prepare a statement based on parse tree. Begin a transaction if necessary.
+ * If the query is not issued in a transaction (when the query type is not 'BEGIN'
+ * and the tcop_txn_state is empty), Peloton will wrap it in a transaction and
+ * begin a transaction here. It is a single_stmt transaction.
+ * Otherwise, the query is issued inside a transaction by the client. It is a multi_stmt transaction.
+ */
 std::shared_ptr<Statement> TrafficCop::PrepareStatement(
-    const std::string &statement_name, const std::string &query_string,
-    UNUSED_ATTRIBUTE std::string &error_message,
+    const std::string &stmt_name, const std::string &query_string,
```

Comment:
I am wondering if we still need the query string here if we already do the parsing in the protocol handler?

Reply:
I think Haoxiang has a PR that fix this problem. Just keep it in this PR.